manifest.go

```diff
@@ -39,12 +45,59 @@ type rawProject struct {
 	Source   string `toml:"source,omitempty"`
 }
 
+func validateManifest(b *bytes.Buffer) error {
+	// Load the TomlTree from reader
+	tree, err := toml.Load(b.String())
+	if err != nil {
+		return errors.Wrap(err, "Unable to load TomlTree from string")
+	}
+	// Convert tree to a map
+	manifest := tree.ToMap()
+
+	// Look for unknown fields and raise warnings
+	for prop, val := range manifest {
+		isValidField := false
+		for _, v := range ManifestFields {
+			if prop == v {
+				isValidField = true
+
+				// For metadata, check if it is a map type
+				if prop == "metadata" {
+					isValidMetadata := false
+					if metadata, ok := val.([]interface{}); ok {
+						for _, v := range metadata {
+							if reflect.TypeOf(v).Kind() == reflect.Map {
+								isValidMetadata = true
+							} else {
+								isValidMetadata = false
+								break
+							}
+						}
+					}
+					if !isValidMetadata {
+						internal.Logf("WARNING: metadata should consist of key-value pairs")
```

Comment:
We're probably going to end up wanting to reuse this func elsewhere, in places where it's not a good idea to log directly. So, instead of directly logging the issues, could we switch to a `([]error, error)` return type, and record all the problems we detect as items in the first slice?